File: backend/FleetMind.Api/Models/Notification.cs
using System;
using FleetMind.Api.Models.Common;

namespace FleetMind.Api.Models;

public class Notification : BaseEntity
{
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
    
    public string Type { get; set; } = null!;
    public string Title { get; set; } = null!;
    public string Message { get; set; } = null!;
    
    public string? RelatedEntityName { get; set; }
    public Guid? RelatedEntityId { get; set; }
    
    public bool IsRead { get; set; }
    public DateTime? ReadAt { get; set; }
}

File: frontend/src/types/notification.ts
import { PaginationQuery } from './pagination';

export interface Notification {
  id: string;
  type: string;
  title: string;
  message: string;
  relatedEntityName: string | null;
  relatedEntityId: string | null;
  isRead: boolean;
  readAt: string | null;
  createdAt: string;
}

export interface NotificationQuery extends PaginationQuery {
  unreadOnly?: boolean;
  type?: string;
}
